diff --git a/inflect/__init__.py b/inflect/__init__.py
index e5ab3f8..fe9ce3a 100644
--- a/inflect/__init__.py
+++ b/inflect/__init__.py
@@ -60,6 +60,7 @@ import itertools
 import re
 from numbers import Number
 from typing import (
+    TYPE_CHECKING,
     Any,
     Callable,
     Dict,
@@ -74,12 +75,9 @@ from typing import (
 )
 
 from more_itertools import windowed_complete
-from pydantic import Field
+from typeguard import typechecked
 from typing_extensions import Annotated, Literal
 
-from .compat.pydantic import same_method
-from .compat.pydantic1 import validate_call
-
 
 class UnknownClassicalModeError(Exception):
     pass
@@ -2021,10 +2019,25 @@ class Words(str):
         self.last = self.split_[-1]
 
 
-Word = Annotated[str, Field(min_length=1)]
 Falsish = Any  # ideally, falsish would only validate on bool(value) is False
 
 
+_STATIC_TYPE_CHECKING = TYPE_CHECKING
+# ^-- Workaround for typeguard AST manipulation:
+#     https://github.com/agronholm/typeguard/issues/353#issuecomment-1556306554
+
+if _STATIC_TYPE_CHECKING:  # pragma: no cover
+    Word = Annotated[str, "String with at least 1 character"]
+else:
+
+    class _WordMeta(type):  # Too dynamic to be supported by mypy...
+        def __instancecheck__(self, instance: Any) -> bool:
+            return isinstance(instance, str) and len(instance) >= 1
+
+    class Word(metaclass=_WordMeta):  # type: ignore[no-redef]
+        """String with at least 1 character"""
+
+
 class engine:
     def __init__(self) -> None:
         self.classical_dict = def_classical.copy()
@@ -2046,7 +2059,7 @@ class engine:
     def _number_args(self, val):
         self.__number_args = val
 
-    @validate_call
+    @typechecked
     def defnoun(self, singular: Optional[Word], plural: Optional[Word]) -> int:
         """
         Set the noun plural of singular to plural.
@@ -2058,7 +2071,7 @@ class engine:
         self.si_sb_user_defined.extend((plural, singular))
         return 1
 
-    @validate_call
+    @typechecked
     def defverb(
         self,
         s1: Optional[Word],
@@ -2083,7 +2096,7 @@ class engine:
         self.pl_v_user_defined.extend((s1, p1, s2, p2, s3, p3))
         return 1
 
-    @validate_call
+    @typechecked
     def defadj(self, singular: Optional[Word], plural: Optional[Word]) -> int:
         """
         Set the adjective plural of singular to plural.
@@ -2094,7 +2107,7 @@ class engine:
         self.pl_adj_user_defined.extend((singular, plural))
         return 1
 
-    @validate_call
+    @typechecked
     def defa(self, pattern: Optional[Word]) -> int:
         """
         Define the indefinite article as 'a' for words matching pattern.
@@ -2104,7 +2117,7 @@ class engine:
         self.A_a_user_defined.extend((pattern, "a"))
         return 1
 
-    @validate_call
+    @typechecked
     def defan(self, pattern: Optional[Word]) -> int:
         """
         Define the indefinite article as 'an' for words matching pattern.
@@ -2131,7 +2144,7 @@ class engine:
         """
         return
 
-    @validate_call
+    @typechecked
     def ud_match(self, word: Word, wordlist: Sequence[Optional[Word]]) -> Optional[str]:
         for i in range(len(wordlist) - 2, -2, -2):  # backwards through even elements
             mo = re.search(rf"^{wordlist[i]}$", word, re.IGNORECASE)
@@ -2271,7 +2284,7 @@ class engine:
 
     # 0. PERFORM GENERAL INFLECTIONS IN A STRING
 
-    @validate_call
+    @typechecked
     def inflect(self, text: Word) -> str:
         """
         Perform inflections in a string.
@@ -2348,7 +2361,7 @@ class engine:
         else:
             return "", "", ""
 
-    @validate_call
+    @typechecked
     def plural(self, text: Word, count: Optional[Union[str, int, Any]] = None) -> str:
         """
         Return the plural of text.
@@ -2372,7 +2385,7 @@ class engine:
         )
         return f"{pre}{plural}{post}"
 
-    @validate_call
+    @typechecked
     def plural_noun(
         self, text: Word, count: Optional[Union[str, int, Any]] = None
     ) -> str:
@@ -2393,7 +2406,7 @@ class engine:
         plural = self.postprocess(word, self._plnoun(word, count))
         return f"{pre}{plural}{post}"
 
-    @validate_call
+    @typechecked
     def plural_verb(
         self, text: Word, count: Optional[Union[str, int, Any]] = None
     ) -> str:
@@ -2417,7 +2430,7 @@ class engine:
         )
         return f"{pre}{plural}{post}"
 
-    @validate_call
+    @typechecked
     def plural_adj(
         self, text: Word, count: Optional[Union[str, int, Any]] = None
     ) -> str:
@@ -2438,7 +2451,7 @@ class engine:
         plural = self.postprocess(word, self._pl_special_adjective(word, count) or word)
         return f"{pre}{plural}{post}"
 
-    @validate_call
+    @typechecked
     def compare(self, word1: Word, word2: Word) -> Union[str, bool]:
         """
         compare word1 and word2 for equality regardless of plurality
@@ -2461,15 +2474,13 @@ class engine:
         >>> compare('egg', '')
         Traceback (most recent call last):
         ...
-        pydantic...ValidationError: ...
-        ...
-          ...at least 1 character...
+        typeguard.TypeCheckError:...is not an instance of inflect.Word
         """
         norms = self.plural_noun, self.plural_verb, self.plural_adj
         results = (self._plequal(word1, word2, norm) for norm in norms)
         return next(filter(None, results), False)
 
-    @validate_call
+    @typechecked
     def compare_nouns(self, word1: Word, word2: Word) -> Union[str, bool]:
         """
         compare word1 and word2 for equality regardless of plurality
@@ -2485,7 +2496,7 @@ class engine:
         """
         return self._plequal(word1, word2, self.plural_noun)
 
-    @validate_call
+    @typechecked
     def compare_verbs(self, word1: Word, word2: Word) -> Union[str, bool]:
         """
         compare word1 and word2 for equality regardless of plurality
@@ -2501,7 +2512,7 @@ class engine:
         """
         return self._plequal(word1, word2, self.plural_verb)
 
-    @validate_call
+    @typechecked
     def compare_adjs(self, word1: Word, word2: Word) -> Union[str, bool]:
         """
         compare word1 and word2 for equality regardless of plurality
@@ -2517,7 +2528,7 @@ class engine:
         """
         return self._plequal(word1, word2, self.plural_adj)
 
-    @validate_call
+    @typechecked
     def singular_noun(
         self,
         text: Word,
@@ -2575,12 +2586,12 @@ class engine:
             return "s:p"
         self.classical_dict = classval.copy()
 
-        if same_method(pl, self.plural) or same_method(pl, self.plural_noun):
+        if pl == self.plural or pl == self.plural_noun:
             if self._pl_check_plurals_N(word1, word2):
                 return "p:p"
             if self._pl_check_plurals_N(word2, word1):
                 return "p:p"
-        if same_method(pl, self.plural) or same_method(pl, self.plural_adj):
+        if pl == self.plural or pl == self.plural_adj:
             if self._pl_check_plurals_adj(word1, word2):
                 return "p:p"
         return False
@@ -3475,7 +3486,7 @@ class engine:
 
     # ADJECTIVES
 
-    @validate_call
+    @typechecked
     def a(self, text: Word, count: Optional[Union[int, str, Any]] = 1) -> str:
         """
         Return the appropriate indefinite article followed by text.
@@ -3556,7 +3567,7 @@ class engine:
 
     # 2. TRANSLATE ZERO-QUANTIFIED $word TO "no plural($word)"
 
-    @validate_call
+    @typechecked
     def no(self, text: Word, count: Optional[Union[int, str]] = None) -> str:
         """
         If count is 0, no, zero or nil, return 'no' followed by the plural
@@ -3594,7 +3605,7 @@ class engine:
 
     # PARTICIPLES
 
-    @validate_call
+    @typechecked
     def present_participle(self, word: Word) -> str:
         """
         Return the present participle for word.
@@ -3613,7 +3624,7 @@ class engine:
 
     # NUMERICAL INFLECTIONS
 
-    @validate_call(config=dict(arbitrary_types_allowed=True))
+    @typechecked
     def ordinal(self, num: Union[Number, Word]) -> str:
         """
         Return the ordinal of num.
@@ -3772,7 +3783,7 @@ class engine:
             num = ONE_DIGIT_WORD.sub(self.unitsub, num, 1)
         return num
 
-    @validate_call(config=dict(arbitrary_types_allowed=True))  # noqa: C901
+    @typechecked
     def number_to_words(  # noqa: C901
         self,
         num: Union[Number, Word],
@@ -3924,7 +3935,7 @@ class engine:
 
     # Join words with commas and a trailing 'and' (when appropriate)...
 
-    @validate_call
+    @typechecked
     def join(
         self,
         words: Optional[Sequence[Word]],
diff --git a/inflect/compat/__init__.py b/inflect/compat/__init__.py
deleted file mode 100644
index e69de29..0000000
diff --git a/inflect/compat/pydantic.py b/inflect/compat/pydantic.py
deleted file mode 100644
index 03bb2a8..0000000
--- a/inflect/compat/pydantic.py
+++ /dev/null
@@ -1,27 +0,0 @@
-class ValidateCallWrapperWrapper:
-    def __new__(cls, wrapped):
-        # bypass wrapper if wrapped method has the fix
-        if wrapped.__class__.__name__ != 'ValidateCallWrapper':
-            return wrapped
-        if hasattr(wrapped, '_name'):
-            return wrapped
-        return super().__new__(cls)
-
-    def __init__(self, wrapped):
-        self.orig = wrapped
-
-    def __eq__(self, other):
-        return self.raw_function == other.raw_function
-
-    @property
-    def raw_function(self):
-        return self.orig.raw_function
-
-
-def same_method(m1, m2) -> bool:
-    """
-    Return whether m1 and m2 are the same method.
-
-    Workaround for pydantic/pydantic#6390.
-    """
-    return ValidateCallWrapperWrapper(m1) == ValidateCallWrapperWrapper(m2)
diff --git a/inflect/compat/pydantic1.py b/inflect/compat/pydantic1.py
deleted file mode 100644
index 8262fdc..0000000
--- a/inflect/compat/pydantic1.py
+++ /dev/null
@@ -1,8 +0,0 @@
-try:
-    from pydantic import validate_call  # type: ignore
-except ImportError:
-    # Pydantic 1
-    from pydantic import validate_arguments as validate_call  # type: ignore
-
-
-__all__ = ['validate_call']
